server/leafnode.go

```diff
@@ -1201,7 +1201,11 @@ func (c *client) processLeafSub(argo []byte) (err error) {
 	}
 
 	// Check if we have a loop.
-	if string(sub.subject) == c.acc.lds {
+	acc := c.acc
+	acc.mu.RLock()
```

Comment:
I would make a func on account. `acc.lds()` and put readlock in there. The just `if string(subject) == acc.lds()`

Reply:
ok, but since account object has lds field, it would have to be `getLds()` or something.

Reply:
Did update, as I mentioned, since field `lds` exits, had to be a different function name.